File: EASSET.Web/Modules/MasterData/SubKegiatan/SubKegiatanColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace EASSET.MasterData.Columns;

[ColumnsScript("MasterData.SubKegiatan")]
[BasedOnRow(typeof(SubKegiatanRow), CheckNames = true)]
public class SubKegiatanColumns
{
    //[EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    //public long Id { get; set; }
    //[EditLink]
    //[DisplayName("Nama Bidang")]
    //public string BidangView { get; set; }
    //[DisplayName("Nama Urusan")]
    //public string UrusanView { get; set; }
    //[DisplayName("Nama Program")]
    //public string ProgView { get; set; }
    [Hidden]
    [DisplayName("Nama Kegiatan"), SortOrder(1, false)]
    public string KegView { get; set; }
    [DisplayName("Nama SubKegiatan"), Width(1000)]
    public string SubKegView { get; set; }
    //public string SubKeg50Id { get; set; }
    //public string SubKeg50Kode { get; set; }
}

File: EASSET.Web/Modules/ServerTypes/MasterData/SubKegiatanColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SubKegiatanRow } from "./SubKegiatanRow";

export interface SubKegiatanColumns {
    KegView: Column<SubKegiatanRow>;
    SubKegView: Column<SubKegiatanRow>;
}

export class SubKegiatanColumns extends ColumnsBase<SubKegiatanRow> {
    static readonly columnsKey = 'MasterData.SubKegiatan';
    static readonly Fields = fieldsProxy<SubKegiatanColumns>();
}